components/MapComponent.tsx

```diff
@@ -1,123 +1,146 @@
-  mapStyle?: StyleProp<ViewStyle>;
   initialRegion: {
-    latitude: number,
-    longitude: number,
-    latitudeDelta: number,
-    longitudeDelta: number,
+    latitude: number
+    longitude: number
+    latitudeDelta: number
+    longitudeDelta: number
   }
-  markers: {
-  key: number,
-  place: string,
-  lat: number,
-  lng: number,
-  pinColor: string
-  }[];
+
+  markers: mapMarker[]
+
   movementSettings: {
-  canScroll: boolean,
-  canZoom: boolean,
-  canRotate: boolean,
-  };
+    canScroll: boolean
+    canZoom: boolean
+    canRotate: boolean
+  }
+
+  extractKey: (event: any) => string
 }
 
 // Map view component itself
-function MapComponent ({initialRegion, markers, containStyle, mapStyle, movementSettings}: Props) {
-  const mapRef = React.useRef(null);
-  const [buttonVisible, setButtonVisible] = useState(true);
-  const [selectedMarker, setSelectedMarker] = useState<Object>();
-  const [currentMarkers, setCurrentMarkers] = useState(markers);
-
-  /*const onLongPress = () => {
-    setCurrentMarkers([...currentMarkers, e.nativeEvent.coordinate]);
-  }*/
-  
+export function MapComponent<T extends mapMarker> ({
+  initialRegion,
+  markers,
+  extractKey,
+  containStyle,
+  mapStyle,
+  movementSettings
+}: Props<T>) {
+  // Map references
+  const mapRef = React.useRef<MapView | null>(null)
+  const [selectedMarker, setSelectedMarker] = useState<String | undefined>()
+
+  const markerData = markers.map((markedMarker) => {
+    const key = extractKey(markedMarker)
+    return {
+      markedMarker,
+      isSelected: key === selectedMarker
+    }
+  })
+
+  // Return the markers so that they appear on the map
   const mapMarkerCreations = () => {
-  return markers.map((report) =>
-    <Marker
-    key = {report.key}
-    title = {report.place}
-    coordinate = {{ latitude: report.lat, longitude: report.lng }}
-    pinColor = {report.pinColor}
-    onPress={() => {onMarkerClick(report.lat, report.lng); setButtonVisible(true); setSelectedMarker(report)}}
-    icon={require('../assets/icon.png')}
-    >
-    </Marker >)
+    return markerData.map((marker) => (
+      <Marker
+        key={marker.markedMarker.key}
+        title={marker.markedMarker.title}
+        coordinate={{
+          latitude: marker.markedMarker.location.latitude,
+          longitude: marker.markedMarker.location.longitude
+        }}
+        pinColor={"blue"}
+        onPress={() => {
+          onMarkerClick(
+            marker.markedMarker.location.latitude,
+            marker.markedMarker.location.longitude
+          )
+          onSelected(marker.markedMarker)
+        }}
+      />
+    ))
   }
 
+  // When clicking on the marker, zoom towards where it is.
   function onMarkerClick (lat: number, long: number) {
     mapRef.current.animateToRegion({
       latitude: lat,
       longitude: long,
       latitudeDelta: initialRegion.latitudeDelta - 0.02,
-      longitudeDelta: initialRegion.longitudeDelta - 0.02,
```

Comment:
Pull “0.02” out into it’s own variable so it’s something we can easily subtitute later, maybe depending on screen size or other factors